StreamSegmentContainerTests failure

**Problem description**
Unit test failure : io.pravega.segmentstore.server.containers.StreamSegmentContainerTests.testExtendedAttributesConditionalUpdates

Stacktrace :
```
java.lang.AssertionError: Unexpected value for attribute 01257154-db51-4582-b998-5ce2da28681c for segment Segment_0 expected:<2> but was:<-9223372036854775808>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at io.pravega.segmentstore.server.containers.StreamSegmentContainerTests.testExtendedAttributesConditionalUpdates(StreamSegmentContainerTests.java:516)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
```
**Problem location**


**Suggestions for an improvement**

--- a/segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentContainerTests.java
+++ b/segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentContainerTests.java
@@ -444,8 +444,8 @@ public class StreamSegmentContainerTests extends ThreadPooledTestSuite {
      */
     @Test
     public void testExtendedAttributesConditionalUpdates() throws Exception {
-        final UUID ea1 = UUID.randomUUID();
-        final UUID ea2 = UUID.randomUUID();
+        final UUID ea1 = new UUID(0, 1);
+        final UUID ea2 = new UUID(0, 2);
         final List<UUID> allAttributes = Stream.of(ea1, ea2).collect(Collectors.toList());
         final TestContainerConfig containerConfig = new TestContainerConfig();
         containerConfig.setSegmentMetadataExpiration(Duration.ofMillis(250));
@@ -508,12 +508,16 @@ public class StreamSegmentContainerTests extends ThreadPooledTestSuite {
 
         Futures.allOf(opFutures).get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
 
-        // 4. Verify results.
+        // 4. Evict the segment from memory, then verify results.
+        localContainer.triggerMetadataCleanup(segmentNames).get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
         for (String segmentName : segmentNames) {
             // Verify all attribute values.
-            SegmentProperties sp = localContainer.getStreamSegmentInfo(segmentName, false, TIMEOUT).join();
+            val attributeValues = localContainer.getAttributes(segmentName, allAttributes, true, TIMEOUT).join();
+            val sp = localContainer.getStreamSegmentInfo(segmentName, false, TIMEOUT).join();
             for (val attributeId : allAttributes) {
-                Assert.assertEquals("Unexpected value for attribute " + attributeId + " for segment " + segmentName,
+                Assert.assertEquals("Unexpected value for non-cached attribute " + attributeId + " for segment " + segmentName,
+                        expectedAttributeValue.get(), (long) attributeValues.getOrDefault(attributeId, Attributes.NULL_ATTRIBUTE_VALUE));
+                Assert.assertEquals("Unexpected value for metadata attribute " + attributeId + " for segment " + segmentName,
                         expectedAttributeValue.get(), (long) sp.getAttributes().getOrDefault(attributeId, Attributes.NULL_ATTRIBUTE_VALUE));
             }
         }

